src/Symfony/Component/HttpFoundation/RequestMatcher/IsJsonRequestMatcher.php

```diff
@@ -23,12 +23,6 @@ class IsJsonRequestMatcher implements RequestMatcherInterface
 {
     public function matches(Request $request): bool
     {
-        try {
-            json_decode($request->getContent(), true, 512, \JSON_BIGINT_AS_STRING | \JSON_THROW_ON_ERROR);
-        } catch (\JsonException) {
-            return false;
-        }
-
-        return true;
+        return json_validate($request->getContent());
```

Comment:
what about the flags?

Reply:
The depth that we used previously was the default. Is was only set explicitly because this is how you could access the `$flags` property before we had named arguments. Neither of the flags that we've used here is accepted by `json_validate()`.

Reply:
`BIGINT_AS_STRING` has no effect if the PHP data structure is not created. And `THROW_ON_ERROR`  is not supported by `json_validate` (as invalid json returns `false`)

Reply:
and the depth is the default one.